File: courses/codes/BoxTransforms.java
import java.util.*;

public class BoxTransforms {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        
        int[] a = new int[n];
        int[] b = new int[n];

        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        for (int i = 0; i < n; i++) {
            b[i] = sc.nextInt();
        }

        List<int[]> keyPositions = new ArrayList<>();
        int target = b[0];
        int A_index = 0;
        int i = 0;

        while (i < n) {
            while (i < n && b[i] == target) {
                i++;
            }
            while (A_index < n && a[A_index] != target) {
                A_index++;
            }
            if (A_index >= n) {
                System.out.println("NO");
                return;
            }
            keyPositions.add(new int[]{target, A_index});
            if (i < n) {
                target = b[i];
            }
        }

        System.out.println("YES");

        List<int[]> loyalMoves = new ArrayList<>();
        int loyal = 0;
        target = b[0];
        int keyIndex = 0;
        i = 0;

        while (i < n && keyIndex < keyPositions.size()) {
            while (i < n && b[i] == target) {
                i++;
            }
            if (keyPositions.get(keyIndex)[1] > loyal) {
                loyalMoves.add(new int[]{loyal, keyPositions.get(keyIndex)[1]});
            }
            if (i < n) {
                target = b[i];
            }
            loyal = i;
            keyIndex++;
        }

        List<int[]> royalMoves = new ArrayList<>();
        int royal = n - 1;
        target = b[n - 1];
        keyIndex = keyPositions.size() - 1;
        i = n - 1;

        while (i >= 0 && keyIndex >= 0) {
            while (i >= 0 && b[i] == target) {
                i--;
            }
            if (keyPositions.get(keyIndex)[1] < royal) {
                royalMoves.add(new int[]{keyPositions.get(keyIndex)[1], royal});
            }
            if (i >= 0) {
                target = b[i];
            }
            royal = i;
            keyIndex--;
        }

        System.out.println(loyalMoves.size() + royalMoves.size());
        for (int[] move : loyalMoves) {
            System.out.println("L " + (move[0] + 1) + " " + (move[1] + 1));
        }
        for (int[] move : royalMoves) {
            System.out.println("R " + (move[0] + 1) + " " + (move[1] + 1));
        }
    }
}

File: courses/codes/cclv1-sr-class/box_transforms.py
import sys

def main():
    n = int(sys.stdin.readline().strip())
    a = list(map(int, sys.stdin.readline().split()))
    b = list(map(int, sys.stdin.readline().split()))

    key_positions = []
    target = b[0]
    A_index = 0

    i = 0
    while i < n:
        while i < n and b[i] == target:
            i += 1
        while A_index < n and a[A_index] != target:
            A_index += 1
        if A_index >= n:
            print("NO")
            return
        key_positions.append((target, A_index))
        if i < n:
            target = b[i]

    print("YES")

    loyal_moves = []
    loyal = 0
    target = b[0]
    key_index = 0

    i = 0
    while i < n and key_index < len(key_positions):
        while i < n and b[i] == target:
            i += 1
        if key_positions[key_index][1] > loyal:
            loyal_moves.append((loyal, key_positions[key_index][1]))
        if i < n:
            target = b[i]
        loyal = i
        key_index += 1

    royal_moves = []
    royal = n - 1
    target = b[n - 1]
    key_index = len(key_positions) - 1

    i = n - 1
    while i >= 0 and key_index >= 0:
        while i >= 0 and b[i] == target:
            i -= 1
        if key_positions[key_index][1] < royal:
            royal_moves.append((key_positions[key_index][1], royal))
        if i >= 0:
            target = b[i]
        royal = i
        key_index -= 1

    print(len(loyal_moves) + len(royal_moves))
    for l, r in loyal_moves:
        print(f"L {l + 1} {r + 1}")
    for l, r in royal_moves:
        print(f"R {l + 1} {r + 1}")

if __name__ == "__main__":
    main()